woocommerce/woocommerce-ios
# Add in-progress UI and show it while updating a Product remotely

Resolves #1671 
Resolves #1668 

## Changes

- Created `InProgressViewController` for an in-progress UI with configurable title and message specified in `InProgressViewProperties`
- In `ProductFormViewController`, presented `InProgressViewController` while updating the Product remotely
  - Set the in-progress view controller's `modalPresentationStyle` to `.overCurrentContext` for below iOS 13, otherwise the presented view controller with a transparent background color will turn black as in this StackOverflow post: https://stackoverflow.com/questions/33104743/swift-modal-view-controller-with-transparent-background/33106533
- In `ProductFormViewController`, updated the navigation bar button item title from "Done" to "Update" as a somewhat related quick update for #1668 

## Testing

- Launch the app
- Go to Products tab
- Tap on a simple Product
- Tap "Edit"
- Tap "Done" --> the in-progress UI should be shown and then dismissed when the Product is updated remotely

## Example screenshots

@kyleaparker I used the `UIBlurEffect` dark style like in the Materials (some of the Material styles are iOS 13+ only). The title has white color, and the spinner and message both have Muriel gray shade 10 for both Light and Dark modes. Lemme know if there are better colors for these views!

State | Dark | Light
-- | -- | --
Edit Product main screen | ![Simulator Screen Shot - iPhone 8 - 2019-12-30 at 16 26 27](https://user-images.githubusercontent.com/1945542/71574293-1c7c9d80-2b23-11ea-93d0-1dae78cc6dd7.png) | ![Simulator Screen Shot - iPhone 8 - 2019-12-30 at 16 26 43](https://user-images.githubusercontent.com/1945542/71574294-1c7c9d80-2b23-11ea-9ca7-7d0f83c86854.png)
In progress | ![Simulator Screen Shot - iPhone 11 - 2020-01-03 at 11 23 19](https://user-images.githubusercontent.com/1945542/71705968-15080d80-2e1d-11ea-9261-07f86be0d6cd.png) | ![Simulator Screen Shot - iPhone 11 - 2020-01-03 at 11 23 06](https://user-images.githubusercontent.com/1945542/71705967-15080d80-2e1d-11ea-88b8-fbc35f72fcbe.png)


Update release notes:
